runtime/queries/metricsview_rows.go

```diff
@@ -153,7 +207,21 @@ func (q *MetricsViewRows) buildMetricsRowsSQL(mv *runtimev1.MetricsView, dialect
 		q.Limit = 100
 	}
 
-	sql := fmt.Sprintf("SELECT * FROM %q WHERE %s %s LIMIT %d OFFSET %d",
+	selectColumns := []string{"*"}
+
+	if timeRollupColumnName != "" {
+		if mv.TimeDimension == "" || q.TimeGranularity == runtimev1.TimeGrain_TIME_GRAIN_UNSPECIFIED {
+			panic("timeRollupColumnName is set, but time dimension info is missing")
+		}
+
+		rollup := fmt.Sprintf("date_trunc('%s', %s) AS %s", convertToDateTruncSpecifier(q.TimeGranularity), safeName(mv.TimeDimension), safeName(timeRollupColumnName))
+
+		// Prepend the rollup column
+		selectColumns = append([]string{rollup}, selectColumns...)
```

Comment:
why not just `[]string{rollup, "*"}` ?

Reply:
Just in case we need to conditionally inject other columns in the future, someone might miss that there's a hard reset. 

Probably a bit premature abstraction